core/src/test/java/com/ibm/wala/core/tests/cha/AnalysisScopeTest.java

```diff
@@ -55,4 +71,61 @@ public void testBaseScope() throws IOException, ClassHierarchyException {
                 ClassLoaderReference.Application, "Ljava/awt/AlphaComposite")),
         "found unexpected class");
   }
+
+  @Test
+  public void testToJson() throws IOException {
+    AnalysisScope scope =
+        AnalysisScopeReader.instance.readJavaScope(
+            TestConstants.WALA_TESTDATA,
+            new FileProvider().getFile("GUIExclusions.txt"),
+            AnalysisScopeTest.class.getClassLoader());
+    Gson gson = new Gson();
+    Type type = new TypeToken<LinkedHashMap<String, Object>>() {}.getType();
+    LinkedHashMap<String, Object> map = gson.fromJson(scope.toJson(), type);
+    assertEquals(
+        List.of("java\\/awt\\/.*", "javax\\/swing\\/.*", "sun\\/awt\\/.*", "sun\\/swing\\/.*"),
+        map.get("Exclusions"));
+
+    @SuppressWarnings("unchecked")
+    Map<String, List<String>> loaders = (Map<String, List<String>>) map.get("Loaders");
+    Set<String> loaderKeys =
+        new HashSet<>(List.of("Primordial", "Extension", "Application", "Synthetic"));
+    assertEquals(loaders.keySet(), loaderKeys);
+    assertEquals(2, loaders.get("Primordial").size());
+    assertThat(loaders.get("Primordial"), hasItem("Nested Jar File:primordial.jar.model"));
+    assertEquals(1, loaders.get("Application").size());
+    assertThat(
+        loaders.get("Application").get(0), containsString("com.ibm.wala.core.testdata_1.0.0.jar"));
+    assertEquals(0, loaders.get("Extension").size());
+    assertEquals(0, loaders.get("Synthetic").size());
+  }
+
+  @Test
+  public void testToJsonCustom() throws IOException {
+    AnalysisScope scope;
+    scope = AnalysisScope.createJavaAnalysisScope();
+    String[] stdlibs = WalaProperties.getJ2SEJarFiles();
+    Arrays.sort(stdlibs);
+    for (String stdlib : stdlibs) {
+      scope.addToScope(ClassLoaderReference.Primordial, new JarFile(stdlib));
+      scope.addToScope(ClassLoaderReference.Application, new JarFile(stdlib));
+    }
+    scope.setExclusions(null);
+    Gson gson = new Gson();
+    Type type = new TypeToken<LinkedHashMap<String, Object>>() {}.getType();
+    LinkedHashMap<String, Object> map = gson.fromJson(scope.toJson(), type);
+    assertEquals(List.of(), map.get("Exclusions"));
+
+    @SuppressWarnings("unchecked")
+    Map<String, List<String>> loaders = (Map<String, List<String>>) map.get("Loaders");
+    Set<String> loaderKeys =
+        new HashSet<>(List.of("Primordial", "Extension", "Application", "Synthetic"));
+    assertEquals(loaders.keySet(), loaderKeys);
+    assertEquals(stdlibs.length, loaders.get("Primordial").size());
+    assertTrue(loaders.get("Primordial").get(0).contains("/Contents/Home/jmods/java.base.jmod"));
```

Comment:
This is still wrong, and tests are still failing.  You need to take the relevant strings from the `stdlibs` array for this comparison, rather than just doing `get(0)` and comparing to a constant string.